src/pages/index.js

```diff
@@ -68,51 +98,138 @@ cardPreviewPopup.close();
 /*                                Card Section                                */
 /* -------------------------------------------------------------------------- */
 
+
 // This function creates a new card
-function createCard(data) {
-    const cardElement = new Card({ data, handleImageClick: (imageData) => {
-        cardPreviewPopup.open(imageData);
+function createCard(data, userId) {
+    const cardElement = new Card({ 
+        data,
+        handleImageClick: (imageData) => {
 
-    }}, selectors.cardTemplate);
+            // Open image popup on click
+            cardPreviewPopup.open(imageData);
+        },
+        handleDeleteClick: () => {
+            
+            // Create the delete popup instance
+            const deletePopup = new PopupWithConfirm(selectors.deletePopup, () => {
 
-    return cardElement.getView();
-}
+                // Handle card deletion
+                api.deleteCard(data._id)
+                .then(() => {
+
+                    // Close the confirmation popup
+                    deletePopup.close();
 
-// Create a section of cards
-const cardSection = new Section(
-    {
-        items: initialCards,
-        renderer: (data) => {
+                    // Delete the card on the page
+                    cardElement.handleDeleteCard();
+                })
+            });
 
-            // Create a new card
-            const cardElement = createCard(data);
+            // Open confirmation popup on click
+            deletePopup.open();
 
-            // Display each card
-            cardSection.addItem(cardElement);
+            // Set the event listeners for the confirmation popup
+            deletePopup.setEventListeners();
         },
-    },
-    selectors.cardsList
-);
+        handleLikeClick: () => {
+
+            // If the user has liked the card, remove the like; vice versa
+            if (cardElement.isLiked()) {
+
+                // Remove like from the server if user has already liked the card
+                api.removeLike(data._id)
+                .then((card) => {
+
+                    // Update like count
+                    cardElement.updateLikeCount(card.likes);
+
+                    // Deactivate like icon
+                    cardElement.removeLikeIcon();
```

Comment:
You should have only one method for updating likes

Reply:
Done!